vms/avm/txs/txstest/builder.go

```diff
@@ -0,0 +1,218 @@
+	payload []byte,
+	owners []*secp256k1fx.OutputOwners,
+	kc *secp256k1fx.Keychain,
+	changeAddr ids.ShortID,
+) (*txs.Tx, error) {
+	xBuilder, xSigner := b.builders(kc)
+
+	utx, err := xBuilder.NewOperationTxMintNFT(
+		assetID,
+		payload,
+		owners,
+		options(changeAddr, nil /*memo*/)...,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed minting NFTs: %w", err)
+	}
+
+	return signer.SignUnsigned(context.Background(), xSigner, utx)
+}
+
+func (b *Builder) MintFTs(
+	outputs map[ids.ID]*secp256k1fx.TransferOutput,
+	kc *secp256k1fx.Keychain,
+	changeAddr ids.ShortID,
+) (*txs.Tx, error) {
+	xBuilder, xSigner := b.builders(kc)
+
+	utx, err := xBuilder.NewOperationTxMintFT(
+		outputs,
+		options(changeAddr, nil /*memo*/)...,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed minting FTs: %w", err)
+	}
+
+	return signer.SignUnsigned(context.Background(), xSigner, utx)
+}
+
+func (b *Builder) Operation(
+	ops []*txs.Operation,
+	kc *secp256k1fx.Keychain,
+	changeAddr ids.ShortID,
+) (*txs.Tx, error) {
+	xBuilder, xSigner := b.builders(kc)
+
+	utx, err := xBuilder.NewOperationTx(
+		ops,
+		options(changeAddr, nil /*memo*/)...,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed building operation tx: %w", err)
+	}
+
+	return signer.SignUnsigned(context.Background(), xSigner, utx)
+}
+
+func (b *Builder) ImportTx(
+	sourceChain ids.ID,
+	to ids.ShortID,
+	kc *secp256k1fx.Keychain,
+) (*txs.Tx, error) {
+	xBuilder, xSigner := b.builders(kc)
+
+	outOwner := &secp256k1fx.OutputOwners{
+		Locktime:  0,
+		Threshold: 1,
+		Addrs:     []ids.ShortID{to},
+	}
+
+	utx, err := xBuilder.NewImportTx(
+		sourceChain,
+		outOwner,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed building import tx: %w", err)
+	}
+
+	return signer.SignUnsigned(context.Background(), xSigner, utx)
+}
+
+func (b *Builder) ExportTx(
+	destinationChain ids.ID,
+	to ids.ShortID,
+	exportedAssetID ids.ID,
+	exportedAmt uint64,
+	kc *secp256k1fx.Keychain,
+	changeAddr ids.ShortID,
+) (*txs.Tx, error) {
+	xBuilder, xSigner := b.builders(kc)
+
+	outputs := []*avax.TransferableOutput{{
+		Asset: avax.Asset{ID: exportedAssetID},
+		Out: &secp256k1fx.TransferOutput{
+			Amt: exportedAmt,
+			OutputOwners: secp256k1fx.OutputOwners{
+				Locktime:  0,
+				Threshold: 1,
+				Addrs:     []ids.ShortID{to},
+			},
+		},
+	}}
+
+	utx, err := xBuilder.NewExportTx(
+		destinationChain,
+		outputs,
+		options(changeAddr, nil /*memo*/)...,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed building export tx: %w", err)
+	}
+
+	return signer.SignUnsigned(context.Background(), xSigner, utx)
+}
+
+func (b *Builder) builders(kc *secp256k1fx.Keychain) (builder.Builder, signer.Signer) {
+	var (
+		addrs = kc.Addresses()
+		wa    = &walletBackendAdapter{
+			b:     b.backend,
+			addrs: addrs,
+		}
+		builder = builder.New(addrs, b.ctx, wa)
+		signer  = signer.New(kc, wa)
+	)
+	return builder, signer
+}
+
+func options(changeAddr ids.ShortID, memo []byte) []common.Option {
```

Comment:
I actually think it might be cleaner to reduce this function's scope to just return the `WithChangeOwner` option and remove `WithMemo`, or to delete this function entirely. Explicitly passing `nil` for memo where it's not needed looks like an anti-pattern to me since this is already an optional parameter via `common.Option`.

Reply:
Not all memo instances are nil. Specifically BaseTx has some tests with non nil memo

Reply:
Cleaned up